Update to Go 1.20

`rand.Seed` has been deprecated since Go 1.20.

See: https://go.dev/doc/go1.20#minor_library_changes

It can be detected by staticcheck.

```
$ staticcheck ./...
main.go:59:2: rand.Seed has been deprecated since Go 1.20 and an alternative has been available since Go 1.0: Programs that call Seed and then expect a specific sequence of results from the global random source (using functions such as Int) can be broken when a dependency changes how much it consumes from the global random source. To avoid such breakages, programs that need a specific result sequence should use NewRand(NewSource(seed)) to obtain a random generator that other packages cannot access.  (SA1019)
```

--- a/go.mod
+++ b/go.mod
@@ -1,3 +1,3 @@
 module github.com/GoCon/2023-Sponsors
 
-go 1.19
+go 1.20

--- a/main.go
+++ b/main.go
@@ -56,7 +56,8 @@ func run() error {
 	}
 
 	seed := time.Now().UnixNano()
-	rand.Seed(seed)
+	rnd := rand.New(rand.NewSource(seed))
+
 	fmt.Println("Seed is", seed)
 	fmt.Println()
 
@@ -64,7 +65,7 @@ func run() error {
 
 	// Platinum "Go"ld sponsor
 	fmt.Println(`==== Platinum "Go"ld sponsor ====`)
-	rand.Shuffle(len(applicants[planPlaTinum]), func(i, j int) {
+	rnd.Shuffle(len(applicants[planPlaTinum]), func(i, j int) {
 		applicants[planPlaTinum][i], applicants[planPlaTinum][j] = applicants[planPlaTinum][j], applicants[planPlaTinum][i]
 	})
 	printCompany(applicants[planPlaTinum][0].company, 1 * time.Second)
@@ -78,7 +79,7 @@ func run() error {
 
 	// "Go"ld sponsor
 	fmt.Println(`==== "Go"ld sponsor ====`)
-	rand.Shuffle(len(applicants[planGold]), func(i, j int) {
+	rnd.Shuffle(len(applicants[planGold]), func(i, j int) {
 		applicants[planGold][i], applicants[planGold][j] = applicants[planGold][j], applicants[planGold][i]
 	})
 	printCompany(applicants[planGold][0].company, 1 * time.Second)
@@ -93,7 +94,7 @@ func run() error {
 
 	// Silver sponsor
 	fmt.Println("==== Silver sponsor ====")
-	rand.Shuffle(len(applicants[planSilver]), func(i, j int) {
+	rnd.Shuffle(len(applicants[planSilver]), func(i, j int) {
 		applicants[planSilver][i], applicants[planSilver][j] = applicants[planSilver][j], applicants[planSilver][i]
 	})
 	for _, a := range applicants[planSilver] {
@@ -103,7 +104,7 @@ func run() error {
 
 	// Bronze sponsor
 	fmt.Println("==== Bronze sponsor ====")
-	rand.Shuffle(len(applicants[planBronze]), func(i, j int) {
+	rnd.Shuffle(len(applicants[planBronze]), func(i, j int) {
 		applicants[planBronze][i], applicants[planBronze][j] = applicants[planBronze][j], applicants[planBronze][i]
 	})
 	for _, a := range applicants[planBronze] {
@@ -113,7 +114,7 @@ func run() error {
 
 	// Free sponsor
 	fmt.Println("==== Free sponsor ====")
-	rand.Shuffle(len(applicants[planFree]), func(i, j int) {
+	rnd.Shuffle(len(applicants[planFree]), func(i, j int) {
 		applicants[planFree][i], applicants[planFree][j] = applicants[planFree][j], applicants[planFree][i]
 	})
 	for _, a := range applicants[planFree] {

